src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1294,53 +1294,64 @@ public void ShowUpdatePackages(ShowUpdatePackageOptions updatePackageOptions)
             }  
         }
 
-        public void SelectFirstPackage(string packageVersion)
+        public void SelectFirstPackage(NuGetVersion packageVersion)
         {
             _topPanel.SelectFilter(ItemFilter.All);
             SelectedSource = PackageSources.FirstOrDefault(psm => psm.IsAggregateSource);
             EventHandler handler = null;
-            handler = (s, e) =>
+            if (_packageList.IsLoaded)
             {
-                _packageList.LoadItemsCompleted -= handler;
+                SelectPackageUponLoaded(packageVersion);
+            }
 
-                _packageList.UpdateSelectedItem((PackageItemViewModel)_packageList.Items.FirstOrDefault());
-                if (_packageList.SelectedItem == null)
+            else
+            {
+                handler = (s, e) =>
                 {
-                    SelectionChangedEventHandler handle = null;
-                    handle = (send, events) =>
-                    {
-                        _packageList.SelectionChanged -= handle;
-                        FindExactPackageVersion(packageVersion);
-                    };
-                    _packageList.SelectionChanged += handle;
-                }
-                else
+                    _packageList.LoadItemsCompleted -= handler;
+                    SelectPackageUponLoaded(packageVersion);
+                };
+                _packageList.LoadItemsCompleted += handler;
+            }
+        }
+
+        private void SelectPackageUponLoaded(NuGetVersion packageVersion)
+        {
+            _packageList.UpdateSelectedItem((PackageItemViewModel)_packageList.Items.FirstOrDefault());
+            if (_packageList.SelectedItem == null)
+            {
+                SelectionChangedEventHandler handle = null;
+                handle = (send, events) =>
                 {
+                    _packageList.SelectionChanged -= handle;
```

Comment:
I'm still weary of this event handler that fires when the selection is changed. I'd like to know why we need this or see if it can be removed.